src/extension/index.activateGithubAnalyses.ts

```diff
@@ -156,24 +160,36 @@ export function activateGithubAnalyses(disposables: Disposable[], store: Store,
             }
         } else {
             // At this point all the local requirements have been satisfied.
-            // We preemptively show the panel (even before the result as fetched)
+            // We preemptively show the panel (even before the result was fetched)
             // so that the banner is visible.
             await panel.show();
-            await onBranchChanged(repo, gitHeadPath, store);
+
+            if (fullCodeScanningAlert) {
+                // bypass downloading analyses. Instead, use the fullCodeScanningAlert as the analysis to apply.
+                await handleSingleLog(fullCodeScanningAlert);
+
+                // Now that the analysis has been successfully applied, avoid re-applying it in the future.
+                // Make sure to update the config value to '' instead of `undefined` to ensure that the
+                // default value is not used. This default value may have been injected by a codespace.
+                await workspace.getConfiguration('sarif-viewer').update('githubCodeScanningInitialAlert', '', ConfigurationTarget.Global);
```

Comment:
Given that this seems to be internal state, a better storage choice for this might be `globalState` on the extension's [context](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext).

Reply:
Unfortunately, it can't be global state. This needs to be injected as a setting by codespaces when the vm is starting up. There is no way to use global state for that AFAIK.

Reply:
Ah, unfortunate. All right, then.